spark/src/main/java/org/apache/iceberg/spark/source/Writer.java

```diff
@@ -303,66 +305,128 @@ public String toString() {
         }
       }
     }
+
+    private class EncryptedOutputFileFactory implements OutputFileFactory<EncryptedOutputFile> {
```

Comment:
Should this be merged with `AppenderFactory`? That could provide `newAppender(PartitionKey, FileFormat)` and `newAppender(FileFormat)` instead of accepting an `OutputFile`.

Reply:
Agreed.

Reply:
Actually, if we merge it, then the `DataWriter` does not have a handle to the currently open file. Those handles are useful to delete empty files on `closeCurrent()` and also to clean up on `abort()`.

Reply:
Left this as is as per comment above.